File: non-overlapping-intervals/non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[1] - b[1]);
        int prevEnd = intervals[0][1];
        int count = 1;

        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i][0] >= prevEnd) {
                prevEnd = intervals[i][1];
                count++;
            }
        }
        return intervals.length - count;
    }
}

File: 435-non-overlapping-intervals/non-overlapping-intervals.py
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        stack = []
        intervals.sort()
        for interval in intervals:
            while stack and stack[-1][1] > interval[1]:
                stack.pop()
            if not stack or stack[-1][1] <= interval[0]:
                stack.append(interval)
        return len(intervals) - len(stack)